### ISSUE TITLE
Respect AuthenticationScheme

### ISSUE THREAD

**Issue by Steinblock** [blowdart/idunno.Authentication#36]
While it is possible to use a custom scheme with

```
services.AddAuthentication("custom")
            .AddBasic("custom", options => { ... });
```

this is ignored by `idunno.BasicAuthenticationHandler`
and the const value of `BasicAuthenticationHandler._Scheme`
is used instead in `HandleAuthenticateAsync` and `HandleChallengeAsync`.

Since the configured scheme is stored in
`AuthenticationHandler<TOptions>.ClaimsIssuer`
this value should be used instead.

**Comment by blowdart** [comment]
Sorry I've taken so long, I've been trying to finish my masters.

The scheme property in Core isn't what gets sent to the browser, it's just a lookup key in a collection used to trigger which handler to run. ClaimsIssuer is used when you build the identity.

However I see what you want to do, so I've added a new property to the options, HttpAuthenticationScheme which does what you were expecting Scheme to do. I just don't want to overload the way Core treats that Scheme lookup key, and it'd confuse having multiple instances with different realms.

Hopefully that works for you.

**Comment by blowdart** [comment]
You should be able to grab the latest build assets to test it, or look at https://github.com/blowdart/idunno.Authentication/commit/78d32cd7c3e2c2c6a7d3686f7815d69e648e0fc5 where it's completed.

**Comment by Steinblock** [comment]
Thanks,

what I really wanted to achive is the following.

I have a frontend single page application written in vue that does mostly ajax requests.

If I send a request that is not authenticated or the credentials are wrong, the browser shows a basic auth popup.

Since I want to handle the error myself and redirect the user to the login route, this is undesired. 

This logic is build into browsers and I can't just disable this on the client side. Thats why some suggest to changed the authorization to something else (xBasic for example) that will not trigger the popup.

This involves the server sending a `WWW-Authenticate: xBasic realm="..."` response and the client responding with a `Authorization: xBasic ...` request.

However I found a more elegant solution..

I added a property `DisableWWWAuthenticateHeader ` to `BasicAuthenticationOptions`

```
        public bool DisableWWWAuthenticateHeader { get; set; }
```

And I changed the `HandleChallengeAsync` method to not send the `WWW-Authenticate` header if `DisableWWWAuthenticateHeader` is set.

This way I can handle the 401 code in my frontend application without getting a popup.

```
        protected override Task HandleChallengeAsync(AuthenticationProperties properties)
        {
            if (!Request.IsHttps && !Options.AllowInsecureProtocol)
            {
                const string insecureProtocolMessage = "Request is HTTP, Basic Authentication will not respond.";
                Logger.LogInformation(insecureProtocolMessage);
                Response.StatusCode = 500;
                var encodedResponseText = Encoding.UTF8.GetBytes(insecureProtocolMessage);
                Response.Body.Write(encodedResponseText, 0, encodedResponseText.Length);
            }
            else
            {
                Response.StatusCode = 401;
                if (!Options.DisableWWWAuthenticateHeader)
                {
                    var headerValue = Options.HttpAuthenticationScheme + $" realm=\"{Options.Realm}\"";
                    Response.Headers.Append(HeaderNames.WWWAuthenticate, headerValue);
                }
            }

            return Task.CompletedTask;
        }
```

**Comment by blowdart** [comment]
Oh that is cleaner. OK, I'll add that instead this morning once I've had coffee :)

**Comment by blowdart** [comment]
Done, give that a whirl :)

**Comment by blowdart** [comment]
@Steinblock if this works for you I'll publish an update this week

**Comment by blowdart** [comment]
Fixed in https://github.com/blowdart/idunno.Authentication/commit/8303c7983989cc029b53555a9fe192aab26d0343

**Comment by blowdart** [comment]
OK all done and published in the [2.2.0](https://www.nuget.org/packages/idunno.Authentication.Basic/2.2.0) package.
